### ISSUE TITLE
Deployment failure on fresh device

### ISSUE THREAD

**Issue by DanConwayDev** [fort-nix/nix-bitcoin#590]
I'm getting the following error on step 7 of  [section '4. Deploy with krops' of the install guide](https://github.com/fort-nix/nix-bitcoin/blob/master/docs/install.md#4-deploy-with-krops). 
```
building Nix...
building the system configuration...
Skipping "/boot/EFI/systemd/systemd-bootx64.efi", since newer boot loader version in place already.
updating systemd-boot from 251.10 to 251.7
Skipping "/boot/EFI/BOOT/BOOTX64.EFI", since newer boot loader version in place already.
Traceback (most recent call last):
  File "/nix/store/ngi744zg26bidgf4mhjcmh14k7vz5ikb-systemd-boot", line 315, in <module>
    main()
  File "/nix/store/ngi744zg26bidgf4mhjcmh14k7vz5ikb-systemd-boot", line 266, in main
    subprocess.check_call(["/nix/store/sf1qmjj1im2xnmdvcpghq2h5b06z5rz2-systemd-251.7/bin/bootctl", "--esp-path=/boot", "update"])
  File "/nix/store/65cp4izx3bllnwqn7c7dhrq9h9gmjkal-python3-3.10.9/lib/python3.10/subprocess.py", line 369, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/nix/store/sf1qmjj1im2xnmdvcpghq2h5b06z5rz2-systemd-251.7/bin/bootctl', '--esp-path=/boot', 'update']' returned non-zero exit status 1.
warning: error(s) occurred while switching to the new configuration
```
I followed guide closely using the encrypted luks UEFI variation.

I attempted to downgrade systemd in my vanilla nixos install to a version below 251.7.

```
[root@hostname:~]# nix-env -iA systemd -f https://github.com/NixOS/nixpkgs/archive/ee01de29d2f58d56b1be4ae24c24bd91c5380cea.tar.gz
replacing old 'systemd-251.4'
installing 'systemd-251.4'
[root@hostname:~]# nix-env --uninstall systemd-251.10
warning: selector 'systemd-251.10' matched no installed derivations
```

I'm getting the same error so I don't think 251.10 was actually removed. I'm also skeptical that this is the right approach.

I imagine that this is an upstream issue but its not obvious to me which package requires the specific 251.7 version of systemd-boot.

Any advice would be much appreciated. Love the project!

**Comment by erikarvstedt** [comment]
Yes, this is an upstream bug, unrelated to nix-bitcoin.
It seems you're installing over an existing system. Try doing a fresh install instead.
`nix-env -iA systemd` only installs systemd to your user environment, it doesn't change the version of systemd used by your system's bootloader or init stages.

**Comment by DanConwayDev** [comment]
Thanks for your response.

I tried removing /boot simply with:

```
[root@nixos:~]# rm /boot -r
rm: cannot remove '/boot': Device or resource busy
```
then I unmounted, reformatted and remounted the partition at /boot:
```
[root@nixos:~]# umount /dev/sda1

[root@nixos:~]# mkfs.fat /dev/sda1
mkfs.fat 4.2 (2021-01-31)

[root@nixos:~]# mount /dev/sda1 /mnt/boot
mount: /mnt/boot: mount point does not exist.
       dmesg(1) may have more information after failed mount system call.

[root@nixos:~]# rm /boot -r

[root@nixos:~]# mkdir /boot

[root@nixos:~]# mount /dev/sda1 /boot

```
This produced the following error:
```
building Nix...
building the system configuration...
systemd-boot not installed in ESP.
No default/fallback boot loader installed in ESP.
Traceback (most recent call last):
  File "/nix/store/ngi744zg26bidgf4mhjcmh14k7vz5ikb-systemd-boot", line 315, in <module>
    main()
  File "/nix/store/ngi744zg26bidgf4mhjcmh14k7vz5ikb-systemd-boot", line 256, in main
    raise Exception("could not find any previously installed systemd-boot")
Exception: could not find any previously installed systemd-boot
warning: error(s) occurred while switching to the new configuration
```
should I manually copy the files bitcoin-node folder and run ```nixos-install``` or is there a cleverer way recreating boot?

**Comment by erikarvstedt** [comment]
No, installing the bootloader requires `nixos-install`.

**Comment by DanConwayDev** [comment]
I manually copied bitcoin-node folder into /etc/nixos, copied nix-bitcoin folder into /tmp-nix-bitcoin-github-files and repointed the configuration here to the tmp director and ran ```nixos-install```. I got the same error.

I booted from usb, formatted the partitions, ssh'd in, copied the files again and ran ```nixos-install```. That worked but when i tried to deploy from my remote machine with ```nix-shell``` with some some minor configuration tweaks I got the same error again. I couldn't just reinstall from usb every time I needed to update the configuration or upgrade.

I slept on it and then ran ```update-nix-bitcoin``` following yesterday's v0.0.86 release and ```deploy``` with ```nix-shell``` was successful.

Perhaps v0.0.86 no longer included the upstream bug.

Thanks for your help @erikarvstedt!
